### PR TITLE
Add Mscorlib and Calling Into the Runtime document

### PR DESCRIPTION
**Pull request by richlander**
This document describes the strong dependency between mscorlib and the CLR. It is an important document to read if you are reading or writing code in either CLR or mscorlib, in many parts of the CoreCLR repo. It was written by @briangru, one of the original developers of the .NET Base Class Libraries.

This document is another chapter in the "Book of the Runtime" series. See PR #115.

### CODE REVIEW

File: Documentation/mscorlib.md [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+
+### QCall Example - Unmanaged Part
+
+Do not replicate the comments into your actual QCall implementation.
+
+The QCall entrypoint has to be registered in tables in [vm\ecalllist.h][ecalllist] using QCFuncEntry macro. See "Registering your QCall or FCall Method" below.
+
+[ecalllist]: https://github.com/dotnet/coreclr/blob/master/src/vm/ecalllist.h
+
+class FooNative
```
Comment (jaredpar) [inline Documentation/mscorlib.md]: Needs to be indented 4 spaces to get formatted as code

File: Documentation/mscorlib.md [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+
+FCalls allow more flexibility in terms of passing object references around, with a higher code complexity and more opportunities to hang yourself. Additionally, FCall methods must either erect a helper method frame along their common code paths, or for any FCall of non-trivial length, explicitly poll for whether a garbage collection must occur. Failing to do so will lead to starvation issues if managed code repeatedly calls the FCall method in a tight loop, because FCalls execute while the thread only allows the GC to run in a cooperative manner.
+
+FCalls require a lot of glue, too much to describe here. Look at [fcall.h][fcall] for details.
+
+[fcall]: https://github.com/dotnet/coreclr/blob/master/src/vm/fcall.h
+
+### GC Holes, FCall, and QCall
+
+TODO: Publish CLR-Coding-Guide to GitHub
```
Comment (jaredpar) [inline Documentation/mscorlib.md]: Can this line be deleted?

Comment (richlander) [inline Documentation/mscorlib.md]: That document is almost done, too, so it is a reminder to update the link in the right place in the doc.

Comment (jaredpar) [inline Documentation/mscorlib.md]: Ah okay, I thought this was for the existing guidelines.

**Pull request merged**